sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -395,6 +396,29 @@ public void setStatus(Status status) {
     }
   }
 
+  @Override
+  public void recordException(Throwable exception) {
+    Preconditions.checkNotNull(exception, "exception");
+    recordException(exception, clock.now());
+  }
+
+  @Override
+  public void recordException(Throwable exception, long timestamp) {
+    Preconditions.checkNotNull(exception, "exception");
+    Preconditions.checkArgument(timestamp >= 0, "Negative timestamp");
+    Attributes.Builder attributes = Attributes.newBuilder();
+    SemanticAttributes.EXCEPTION_TYPE.set(attributes, exception.getClass().getCanonicalName());
+    if (exception.getMessage() != null) {
+      SemanticAttributes.EXCEPTION_MESSAGE.set(attributes, exception.getMessage());
+    }
+    StringBuilder buffer = new StringBuilder();
+    StringBuilderPrintWriter writer = new StringBuilderPrintWriter(buffer);
+    exception.printStackTrace(writer);
+    SemanticAttributes.EXCEPTION_STACKTRACE.set(attributes, buffer.toString());
+
+    addEvent("exception", attributes.build(), timestamp);
```

Comment:
Should we have a semantic "attribute" constant for this event name, as well? Seems like this event name is special, and canonical, so we should probably put it in the API somewhere.

Reply:
Added a constant